impl-base/src/main/java/org/jboss/shrinkwrap/impl/base/container/ContainerBase.java

```diff
@@ -811,7 +811,7 @@ private T addNestedJarFileResource(final File resource, final ArchivePath target
         String adjustedPath = path;
         final int indexOfExclamationPoint = path.indexOf('!' + File.separator);
         if(indexOfExclamationPoint!=-1){
-            adjustedPath = path.substring(indexOfExclamationPoint + 2, path.length());
+            adjustedPath = path.substring(indexOfExclamationPoint + 2, path.length()).replace(File.separator, "/");
```

Comment:
I put the replace of the separator inside the if-statement. Because this method adds a resource from a JAR, isn't indexOfExclamationPoint always != -1?